firefox_ui_harness/runners/base.py

```diff
@@ -39,9 +48,42 @@
     'focusmanager.testmode': True,
     'geo.provider.testing': True,
     'javascript.options.showInConsole': True,
+    'marionette.logging': False,
     'security.notification_enable_delay': 0,
     'signon.rememberSignons': False,
     'startup.homepage_welcome_url': 'about:blank',
     'toolkit.startup.max_resumed_crashes': -1,
     'toolkit.telemetry.enabled': False,
 }
+
+
+class FirefoxUITestRunner(BaseMarionetteTestRunner):
+    def __init__(self, **kwargs):
+        BaseMarionetteTestRunner.__init__(self, **kwargs)
+
+        if not self.server_root:
+            self.server_root = firefox_ui_tests.resources
+
+        # Set default preferences for Firefox UI tests
+        # Bug 1141129 - Workaround necessary given that self.prefs defaults to None
+        prefs = self.prefs or {}
+        prefs.update(DEFAULT_PREFS)
+        self.prefs = prefs
```

Comment:
So after more tests I cannot keep the code as is! It's totally broken and I already filed bug 1146847 to get the priority order fixed. If we keep it none of the preferences we need for update tests are kept but thrown away. Frankly I don't have the time to check where in Marionette this happens. So I will place a workaround for the e10s mode in here for now. I hope that will be fine.